fe/fe-core/src/main/java/com/starrocks/load/routineload/PulsarProgress.java

```diff
@@ -0,0 +1,161 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.starrocks.load.routineload;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+import com.starrocks.common.Pair;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.util.DebugUtil;
+import com.starrocks.thrift.TPulsarRLTaskProgress;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * this is description of pulsar routine load progress
+ * the data before position was already loaded in StarRocks
+ */
+// {"partitionToBacklogNum": {}}
+public class PulsarProgress extends RoutineLoadProgress {
+    private static final Logger LOG = LogManager.getLogger(PulsarProgress.class);
+
+    // (partition, backlog num)
+    private Map<String, Long> partitionToBacklogNum = Maps.newConcurrentMap();
+    // Initial positions will only be used at first schedule
+    private Map<String, Long> partitionToInitialPosition = Maps.newConcurrentMap();
```

Comment:
What does the initial position in interger type stands for? The initial position set in create stmt is in string type.

Reply:
For pulsar usage from BE side, pulsar takes integer as parameter, so I transform it here.